boards: arm: add initial support for efr32bg22_brd4184b

pepe2k opened the pull request:
This short series adds initial support for the `B` revision of the `Silicon Labs EFR32BG BRD4184` (Thunderboard BG22) development board. Support was tested on real hardware but requires change in `hal_silabs`, provided in separated PR: https://github.com/zephyrproject-rtos/hal_silabs/pull/31

Additional on-board peripherals (same as in case of the `A` revision) are still not supported (missing drivers):

- Silicon Labs Si7021 relative humidity and temperature sensor
- Vishay VEML6035 ambient light sensor
- Knowles SPK0641HT4H-1 MEMS microphones
- TDK InvenSense ICM-20648 6-axis inertial sensor

Board documentation has been updated accordingly.

pepe2k committed:
boards: arm: add initial support for efr32bg22_brd4184b

This adds basic support for 'B' revision of the Silicon Labs EFR32BG
BRD4184 (Thunderboard BG22) board. Due to missing drivers, some of the
on-board peripherals are currently unsupported:

- Silicon Labs Si7021 relative humidity and temperature sensor
- Vishay VEML6035 ambient light sensor
- Knowles SPK0641HT4H-1 MEMS microphones
- TDK InvenSense ICM-20648 6-axis inertial sensor

Signed-off-by: Piotr Dymacz <pepe2k@gmail.com>

pepe2k committed:
boards: arm: doc: update EFR32BG BRD4184 documentation

This updates Silicon Labs EFR32BG BRD4184 (Thunderboard BG22) board
documentation, adding information about the 'B' revision.

Signed-off-by: Piotr Dymacz <pepe2k@gmail.com>

fkokosinski commented on boards/arm/efr32_thunderboard/board.cmake:
```diff
@@ -4,6 +4,10 @@ if(CONFIG_BOARD_EFR32BG22_BRD4184A)
 board_runner_args(jlink "--device=EFR32BG22C224F512IM40" "--reset-after-load")
 include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)
 
+elseif(CONFIG_BOARD_EFR32BG22_BRD4184B)
+board_runner_args(jlink "--device=EFR32BG22C224F512IM40" "--reset-after-load")
+include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)
```
These are a copy of what's defined for BRD4184A. Let's combine them in the if clause, i.e.:
```if(CONFIG_BOARD_EFR32BG22_BRD4184A OR CONFIG_BOARD_EFR32BG22_BRD4184B)```

pepe2k replied:
@fkokosinski ok, done!

fkokosinski commented on boards/arm/efr32_thunderboard/efr32bg22_brd4184b.dts:
```diff
@@ -0,0 +1,83 @@
+/*
```
This file looks identical to `boards/arm/efr32_thunderboard/efr32bg22_brd4184a.dts`. Please separate their common parts into a separate dtsi, which is later included in the board dts.

pepe2k replied:
@fkokosinski ok, done (common parts in `efr32bg22_brd4184.dtsi`).

fkokosinski replied:
Thanks!

Can you modify the dtsi hierarchy so that it's linear? e.g. `efr32bg22_brd4184b.dts` -> `efr32bg22_brd4184.dtsi` -> `thunderboard.dtsi`?

fkokosinski replied:
@pepe2k yes

fkokosinski reviewed (changes requested):
Hi, thanks for the PR!

LGTM in general, just some code duplication issues.

fkokosinski commented:
Also: this PR doesn't modify west manifest file (`west.yml`), so this PR is not pulled: https://github.com/zephyrproject-rtos/hal_silabs/pull/31.

fkokosinski commented:
Hi @pepe2k, thanks for the update! I've merged your PR to the `hal_silabs` repository. Please update the `hal_silabs` revision in `west.yml` to `fbb47a4067c460a2c26917ff34189fb4f32f654f`.

pepe2k commented:
> Hi @pepe2k, thanks for the update! I've merged your PR to the `hal_silabs` repository. Please update the `hal_silabs` revision in `west.yml` to `fbb47a4067c460a2c26917ff34189fb4f32f654f`.

Hi @fkokosinski, thanks! I just rebased and updated `west.yml` as requested.
